### ISSUE TITLE
Separate out sampling repo from eval repo

### ISSUE THREAD

**Issue by reubenharry** [reubenharry/blackjax-benchmarks#2]
Our eventual goal is to be able to run comparisons of samplers on a curated set of models, with minimal effort. This repo currently has three functions: 

1. it provides an implementation of a metric which takes a chain and measures number of steps (or gradients) to low error
2. it provides fully configured samplers (currently just NUTS and the Berkeley microcanonical samplers), to produce such chains
3. it provides a set of models (along with appropriate moments) on which these samplers can be run

It is worth thinking carefully about whether and how we should separate these functions into different repos.

## Eval is sampler dependent

On the face of it, [our current metric](https://github.com/reubenharry/blackjax-benchmarks?tab=readme-ov-file#evaluating-samplers) only needs an array of samples [num_chains, num_samples, num_model_dims], the target density and the relevant ground truth moments, so it would be sensible to have a repo dedicated to this. Samples could then be produced by any sampling library (e.g. Stan).

The one caveat is that certain information from a sampler is also relevant to our evaluation. For instance, any gradient-based sampler should be evaluated not in terms of number of samples to low error, but in terms of number of gradient calls to low error (a more realistic proxy for wall-clock time). 

This means we would want to provide two things: the array of samples, and also a "cost per sample". The latter depends on the sampler that produced the samples.

## Bob's proposal

Here, an arrow indicates a dependency:

<img width="709" alt="Image" src="https://github.com/user-attachments/assets/2e37e6d3-7e59-4aef-8505-39b7cfd50dfd" />

Currently the comparisons and eval repo are the same. If the separation is doable cleanly (see concerns above), then I think this sounds good.

**Comment by bob-carpenter** [comment]
Just confirming that this is indeed what I proposed, which is to focus the contribution of each repo.

* **Comparisons**:  this is just something we'll build to evaluate some things we care about---this isn't going to be a public repo we expect other people to extend.  On the other hand, they might fork it and modify it for their own evals.

* **Eval Repo**: This performs the evaluations of samplers in terms of accuracy.

* **Inference Gym**, **Blackjax**, **JAX**: not our repos---just dependencies.  We'll contribute new models to inference gym and new samplers to Blackjax.

I do not think we want to require a sampler be implemented in Blackjax to be compared.  As this issue says, we just need the draws to do accuracy evaluations.  

I do not think we want to require gradient counts, etc. from samplers being evaluated.  They may not always be easy to get.

This brings up the question of whether we want to include any plotting software, and if so, how are we going to do it.  I really like `plotnine`, which basically clones R's `ggplot2` into Python.

**Comment by reubenharry** [comment]
OK, sounds good. I propose that I do the following shortly:

1. Remove all the blackjax dependencies (i.e. the samplers) from this repo
2. Rename this repo `sampler-evaluation`
3. Update the readme accordingly
4. Make a new repo called `sampler-comparison`, where I put all the packaged samplers

I could also keep everything in this repo, but only package `sampler-evaluation`.

Does either of these seem reasonable?

For plotting, I generally use seaborn (you can see it in action in demo.ipynb), which works similarly to matplotlib. But I like the ggplot philosophy, so I'd be happy with that, although I have no experience using it.

**Comment by bob-carpenter** [comment]
Yes, that sounds totally reasonable.  

I can use anything you'd like for plotting---I just ask GPT to write or explain plotting code to me anyway.

**Comment by reubenharry** [comment]
OK, I have done this. For simplicity (debatable) I have kept everything in this repo (now named sampler-benchmarks), and split it into two packages, `sampler-evaluation` and `sampler-comparison`. I import `sampler-evaluation` in `sampler-comparison`, using a local install (`pip install -e .`) because I haven't yet published the former.

I'm going to use the Python tool `poetry` to build and publish these packages, since it seems like a good modern option.

Note that `sampler-evaluation` still has a folder called `models`, which is where I put additions to inference-gym models (e.g. second moments) that are not yet added to inference-gym. Assuming that inference-gym accepts our contributions, this should eventually disappear.

**Comment by bob-carpenter** [comment]
> OK, I have done this. 

If the issue is done, please close it.
